### PR TITLE
[#1818] Add input builder to LocalRepoSystemTest

### LINKED ISSUE #1818: Use `InputBuilder` for `LocalRepoSystemTest`
**What feature(s) would you like to see in RepoSense?**

In #1746, all the test cases uses CLI inputs like this:

```java
String cliInput = String.format("-r %s %s -s d1 -u %s -o %s -t %s", ...);
```

This can be troublesome and error-prone.

**If possible, describe the solution**

`ConfigSystemTest` on the other hand uses `InputBuilder` which offers methods to craft and build CLI inputs.

```java
InputBuilder inputBuilder = new InputBuilder().addConfig(configFolder)
                .addFormats(formats)
                .addTimezone(TEST_TIME_ZONE)
                .addXXX(...);

String cliInput = inputBuilder.build();
```

### PR DESCRIPTION
**Pull request by lawwm**
Fixes #1818 

## Proposed commit message

```
Currently, all test cases within LocalRepoSystemTest uses
String.format().

Let's change the CLI input of test cases within LocalRepoSystemTest to
use the InputBuilder class.
```

### CODE REVIEW

**Comment by SkyBlaise99** [comment]
There is a 72 characters per line hard limit for Proposed commit message. Just make it a new line at about 70 chars.

File: src/systemtest/java/reposense/LocalRepoSystemTest.java [hunk 1]
```diff
@@ -33,7 +34,7 @@ public class LocalRepoSystemTest {
     private static final String OUTPUT_DIRECTORY = "local-test";
     private static final String TIME_ZONE = "UTC+08";
     private static final Path REPORT_DIRECTORY_PATH = Paths.get(OUTPUT_DIRECTORY, "reposense-report");
-
+    private static final String EARLIEST_COMMIT_DATE = "d1";
```
Comment (SkyBlaise99) [inline src/systemtest/java/reposense/LocalRepoSystemTest.java:37]: You can use `SinceDateArgumentType.FIRST_COMMIT_DATE_SHORTHAND` instead.

File: src/systemtest/java/reposense/LocalRepoSystemTest.java [hunk 2]
```diff
@@ -63,16 +64,29 @@ public static void deleteClonedLocalRepos() throws Exception {
...
-                LOCAL_DIRECTORY_ONE, LOCAL_DIRECTORY_TWO, LAST_COMMIT_DATE, OUTPUT_DIRECTORY, TIME_ZONE);
+        InputBuilder inputBuilder = new InputBuilder()
+                .addRepos(LOCAL_DIRECTORY_ONE, LOCAL_DIRECTORY_TWO)
+                .addSinceDate(EARLIEST_COMMIT_DATE)
+                .addUntilDate(LAST_COMMIT_DATE)
+                .addOutput(Paths.get(OUTPUT_DIRECTORY))
+                .addTimezone(TIME_ZONE);
+
+        String cliInput = inputBuilder.build();
         runTest(cliInput, "LocalRepoSystemTest/testSameFinalDirectory");
```
Comment (SkyBlaise99) [inline src/systemtest/java/reposense/LocalRepoSystemTest.java:75]: A bit weird to have some preprocessing here and then another `String[] args = commandArgs.split(" ");` before `RepoSense.main(args);`.

I suggest an alternative way which is to pass the `inputBuilder` to `runTest()`. You can refer to [this](https://github.com/reposense/RepoSense/pull/1816/files#diff-59400ca2297b765e07badbf3424d7ac8fdb8237d902ea99ad60a581447f9b05cR175) to see how it is done.

File: src/systemtest/java/reposense/LocalRepoSystemTest.java [hunk 3]
```diff
@@ -63,16 +64,29 @@ public static void deleteClonedLocalRepos() throws Exception {
...
     }
 
     @Test
     public void testRelativePathing() {
         String relativePathForTesting = "parent1/../parent1/./test-repo";
```
Comment (SkyBlaise99) [inline src/systemtest/java/reposense/LocalRepoSystemTest.java:80]: Can just inline and remove this variable.

**Review by SkyBlaise99 — CHANGES_REQUESTED** [review]
Good job. Please see the comments for the changes. You can also provide reasoning if you think some parts cannot / should not be changed 😄

**Comment by SkyBlaise99** [comment]
> However, adding quotations marks for a local directory such -o "local-test" fails while -o local-test works. Please inform me if I should revert back to adding the quotation marks and find another workaround.

Use `InputBuilder` with `translateCommandline()` to pass the test cases. I see you have changed to that already, I think you can try to revert back the changes and test it out again.

**Comment by lawwm** [comment]
Thank you, `translateCommandline()` seems to solve the quotation issue.

**Comment by SkyBlaise99** [comment]
No problem. Do remember to resolve all requested changes before asking for a review again.

File: src/systemtest/java/reposense/LocalRepoSystemTest.java (hunk 1 above)
Comment (SkyBlaise99) [reply src/systemtest/java/reposense/LocalRepoSystemTest.java:39]: Remember to remove unused variables.

**Review by SkyBlaise99 — CHANGES_REQUESTED** [review]
Looking good, just remember to remove unused variables.

**Comment by SkyBlaise99** [comment]
Next time after you done resolving the changes, click the button here (should be red for you) and there should be an option to request for review again to alert us. Or you can also @ us here or in Slack to notify us.

image

**Comment by lawwm** [comment]
image
Sorry, missed out on the unused variable. Also, I'm not sure why but the ui doesn't display any changes when i click the `re-request review` button.

**Review by chan-j-d — APPROVED** [review]
LGTM!

**Review by SkyBlaise99 — APPROVED** [review]
LGTM! Great job.

**Review by dcshzj — APPROVED** [review]
LGTM!

**Pull request merged**